pkgs/types/src/lib.zig

```diff
@@ -297,6 +297,33 @@ pub const BeamState = struct {
             },
         };
     }
+
+    /// Process a single slot, backfilling the state_root if zero.
+    /// This prepares the state to be the post-state of the slot.
+    pub fn process_slot(self: *BeamState, allocator: Allocator) !void {
+        // update state root in latest block header if its zero hash
+        // i.e. just after processing the latest block of latest block header
+        // this completes latest block header for parentRoot checks of new block
+        if (std.mem.eql(u8, &self.latest_block_header.state_root, &ZERO_HASH)) {
+            var prev_state_root: [32]u8 = undefined;
+            try ssz.hashTreeRoot(BeamState, self.*, &prev_state_root, allocator);
+            self.latest_block_header.state_root = prev_state_root;
+        }
+    }
+
+    /// Process multiple slots, advancing the state to the target slot.
+    /// This prepares the state to be the pre-state of the target slot.
+    pub fn process_slots(self: *BeamState, allocator: Allocator, target_slot: Slot, logger: anytype) !void {
```

Comment:
When `process_slots` migrated as a member function, I feel the `logger` arg as a function arg is not good. Let logger as a module or struct field seems more readable for me.

Reply:
It seems there is another issue for CachedState, may be separate this change. @bomanaps Yes, I remembered the ssz type can't have non ser/dser field, I am not sure if it is possible a module level logger field(not in the struct but on top of the file) work or not. If not, just keep it as an arg.